main/src/main/scala/sbt/Keys.scala

```diff
@@ -17,6 +17,7 @@ import sbt.BuildSyntax._
 import sbt.Def.ScopedKey
 import sbt.KeyRanks._
 import sbt.internal._
+import sbt.internal.FiniteDurationCodec._
```

Comment:
Could you refresh my memory on where this typeclass instance is being captured?

Reply:
The expansion of the `taskKey` macro passes it to the apply in `TaskKey`, if memory serves.

Reply:
If you want to go the long route you could put this in sjsonnew's BasicProtocol and cut a release.

Reply:
I thought changing BasicJsonProtocol breaks the bincompat because it's a trait.

Reply:
Even if it's not part of BasicJsonProtocol, maybe it does make sense to put these format in sjson-new though.

Reply:
Adding to traits is bincompat in Scala 2.12+.  Are you concerned about sjson-new for Scala 2.11?

I'll PR sjson-new and we can have a look.

Reply:
In most cases, but not all. Weren't there a plugin we broke early in sbt 1.x because we added something to BasicJsonProtocol, and someone was extending it?

Reply:
Not that I'm aware of, or I've since forgotten.